src/libos/src/fs/procfs/pid_inode.rs

```diff
@@ -121,11 +145,53 @@ impl DirProcINode for LockedProcFdDirINode {
             0 => Ok(String::from(".")),
             1 => Ok(String::from("..")),
             i => {
-                // TODO: When to iterate the file table ?
-                Err(FsError::EntryNotFound)
+                let file = self.0.read().unwrap();
+                let main_thread = file
+                    .process_ref
+                    .main_thread()
+                    .ok_or(FsError::EntryNotFound)?;
+                let fds = main_thread.files().lock().unwrap().fds();
+                let fd = fds.iter().nth(i - 2).ok_or(FsError::EntryNotFound)?;
+                Ok(fd.to_string())
             }
         }
     }
+
+    fn iterate_entries(&self, mut ctx: &mut DirentWriterContext) -> vfs::Result<usize> {
+        let file = self.0.read().unwrap();
+        let mut total_written_len = 0;
+        let idx = ctx.pos();
+
+        // Write first two special entries
+        if idx == 0 {
+            let this_inode = file.this.upgrade().unwrap();
+            write_inode_entry!(&mut ctx, ".", &this_inode, &mut total_written_len);
+        }
+        if idx <= 1 {
+            write_inode_entry!(&mut ctx, "..", &file.parent, &mut total_written_len);
+        }
```

Comment:
Can we add a function to replace these code in all the procfs files?

Reply:
I think add a macro is good option